x-pack/plugins/gis/public/shared/layers/sources/ems_tms_source/ems_tms_source.js

```diff
@@ -95,25 +103,33 @@ export class EMSTMSSource extends TMSSource {
   }
 
   async getAttributions() {
-    const service = this._getTMSOptions();
-    const attributions = service.attributionMarkdown.split('|');
-
-    return attributions.map((attribution) => {
-      attribution = attribution.trim();
-      //this assumes attribution is plain markdown link
-      const extractLink = /\[(.*)\]\((.*)\)/;
-      const result = extractLink.exec(attribution);
-      return {
-        label: result ? result[1] : null,
-        url: result ? result[2] : null
-      };
-    });
+    let service;
+    let attributions;
+    try {
```

Comment:
What is this try/catch block going to catch? I don't think its needed since `this._getTMSOptions` does not throw and error and `service.attributionMarkdown.split` should not throw an error. This entire method code be rewritten to the below which I think is much more readable

```
async getAttributions() {
  const service = this._getTMSOptions();
  if (!service || !service.attributions) {
    return '';
  }

  return attributions.map((attribution) => {
    attribution = attribution.trim();
    //this assumes attribution is plain markdown link
    const extractLink = /\[(.*)\]\((.*)\)/;
    const result = extractLink.exec(attribution);
    return {
      label: result ? result[1] : null,
      url: result ? result[2] : null
    };
  });
}
```

Reply:
There was an error getting thrown specific to attributions, otherwise I wouldn't have modified it at all since it's not directly related. I believe it was attempting to create attributions on unresolved layers

Reply:
But where is the error getting thrown? Is it getting thrown by `_getTMSOptions`? This method needs some clean-up. It is very difficult to read and has some problems. For example, if an exception is thrown and `attributions ` is not set when you will call `attributions.map` with will throw an exception. Also, your return returns mixed types. Sometimes it will return an array and sometimes it will return an empty string. It should return an `[]` if there are not attributes or there are problems getting them. The implementation I provided should be fine and I don't think needs to be a try/catch block